src/Persistence/Models/Component.cs

```diff
@@ -39,11 +38,11 @@ public Component(string name, string variant, IControlTemplateStore controlTempl
     public bool AccessAppScope { get; set; }
 
     [YamlMember(Order = 100)]
-    public CustomPropertiesCollection CustomProperties { get; set; } = new();
+    public IList<CustomProperty> CustomProperties { get; set; } = new List<CustomProperty>();
 
     internal override void AfterCreate(Dictionary<string, object?> controlDefinition)
     {
-        if (controlDefinition.TryGetValue<CustomPropertiesCollection>(nameof(CustomProperties), out var customProperties))
+        if (controlDefinition.TryGetValue<List<CustomProperty>>(nameof(CustomProperties), out var customProperties))
```

Comment:
nit: since `CustomProperties` has a public setter, might be safer to cast this to `IList<CustomProperty>`

Reply:
Done